### PR TITLE
Set ACCNR in rt.sh, not detect_machine.sh. Add some comments explaini…

### LINKED ISSUE #1014: Bug in rt.sh jet section when ACCNR is not set
## Description
A recent PR (https://github.com/ufs-community/ufs-weather-model/pull/981) updated the jet section in `rt.sh` to allow setting `ACCNR` instead of using the hardcoded `h-nems`. A fallback/default value `h-nems` was intended in this part:
```
ACCNR=${ACCNR:-h-nems}
PARTITION=xjet
DISKNM=/lfs4/HFIP/h-nems/emc.nemspara/RT
dprefix=${dprefix:-/lfs4/HFIP/$ACCNR/$USER}
```
The problem is that bash gets confused by the `-` in the `h-nems` default value, which results in
```
dprefix=/lfs4/HFIP/nems/$USER
```
and causes `rt.sh` abort with
```
+ ACCNR=nems
+ PARTITION=xjet
+ DISKNM=/lfs4/HFIP/h-nems/emc.nemspara/RT
+ dprefix=/lfs4/HFIP/nems/emc.nemspara
+ STMP=/lfs4/HFIP/nems/emc.nemspara/RT_BASELINE
+ PTMP=/lfs4/HFIP/nems/emc.nemspara/RT_RUNDIRS
+ SCHEDULER=slurm
+ cp fv3_conf/fv3_slurm.IN_jet fv3_conf/fv3_slurm.IN
+ cp fv3_conf/compile_slurm.IN_jet fv3_conf/compile_slurm.IN
+ mkdir -p /lfs4/HFIP/nems/emc.nemspara/RT_BASELINE/emc.nemspara
mkdir: cannot create directory '/lfs4/HFIP/nems': Permission denied
++ echo 'rt.sh error on line 395'
```

### To Reproduce:
Try running `rt.sh` on jet without setting `ACCNR`. Then, oops, look at the bug.

### LINKED ISSUE #1015: Many platforms override ACCNR with a hard-coded value
The giant "if/elif/fi" block in rt.sh overrides the ACCNR to a hard-coded value, overriding the value the user sets before running rt.sh.  This is in part necessary due to detect_machine.sh setting the default to nems, which is usually wrong.  An earlier bug fix did not completely correct the problem. (See also #979, #1014, and #981.)

The detect_machine.sh is run before the rt.sh per-platform default logic, setting the nems variable:

```
# Default account "nems"
export ACCNR=${ACCNR:-nems}
```

Hence to use the account most users require, the rt.sh has to override the detect_machine.sh and user's setting with a hard-coded value like so:

```
ACCNR=GFS-DEV
```

The fix is for detect_machine.sh to NOT set ACCNR. Instead, rt.sh's per-platform block will set the defaults in a way that does not override the user's defaults:

```
ACCNR="${ACCNR:-h-nems}"
```

After the giant "if" block there will be a default setting to ensure the original behavior (ACCNR=nems) if neither the user nor per-platform block sets the value:

```
# If account is unspecified, assume the machine has a "nems"
# accounting code.
export ACCNR="${ACCNR:-nems}"
```

### PR DESCRIPTION
**Pull request by SamuelTrahanNOAA**
# PR Checklist

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [ ] ~~Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.~~

- [ ] ~~New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.~~

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

# Description

The rt.sh ignores the ACCNR value set by the user in the environment because the big per-platform "if/elif/fi" hard-codes it to per-platform values (`ACCNR=GFS-DEV`) instead of allowing a user-specified default (`ACCNR=${ACCNR:-GFS-DEV}`). It has to do this because the detect_machine.sh sets an incorrect default value (nems) so ACCNR is always set by the time it reaches the rt.sh per-platform if/elif/fi block. The fix is to remove the setting from detect_machine.sh and allow a user-specified default in rt.sh. The final default of ACCNR=${ACCNR:-nems} happens after that block to ensure the original behavior of a "nems" default when appropriate.

**Warning: If a user or automated test system was setting ACCNR to an invalid value, then rt.sh will use that value after this PR, and will fail. Make certain your scripts are setting ACCNR properly, or not setting it at all, before testing this!**

### Issue(s) addressed

The first attempt at fixing this (issue #979) is in PR #981, which caused a bug #1014 and didn't fix issues mentioned in #1015.

Resolves #1014 and resolves #1015

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

To avoid wasting CPU time, I have run *minimal* testing, sufficient to ensure the ACCNR is passed properly. To fully trust these changes, we need to run baseline generation and regression testing for all platforms.  This PR should be combined with an answer-changing PR to avoid wasting CPU time. Also note that this may break the github-driven automated testing if it set ACCNR incorrectly.

- [x] hera.intel
- [x] hera.gnu
- [ ] orion.intel
- [ ] cheyenne.intel 
- [ ] cheyenne.gnu
- [ ] gaea.intel 
- [x] jet.intel
- [ ] wcoss_cray
- [ ] wcoss_dell_p3
- [ ] opnReqTest for newly added/changed feature
- [ ] CI

## Dependencies

None.

### CODE REVIEW

**Comment by SamuelTrahanNOAA** [comment]
To make sure nobody misses it, I'm going to repeat this warning from the PR description:

**Warning: If a user or automated test system was setting ACCNR to an invalid value, then rt.sh will use that value after this PR, and will fail. Make certain your scripts are setting ACCNR properly, or not setting it at all, before testing this!**

**Comment by SamuelTrahanNOAA** [comment]
Also, an answer-changing PR should be merged into this branch as soon as possible so the system can be fully tested. I only tested enough to ensure the ACCNR is passed correctly.

**Comment by MinsukJi-NOAA** [comment]
> Also, an answer-changing PR should be merged into this branch as soon as possible so the system can be fully tested. I only tested enough to ensure the ACCNR is passed correctly.

I will pull this PR into #1009, which is the current one in the commit queue.

**Comment by SamuelTrahanNOAA** [comment]
@MinsukJi-NOAA Your branch breaks Jet support, so I'll need something else to test against.

```
CMake Error at CMakeLists.txt:111 (find_package):
  By not providing "FindNetCDF.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "NetCDF", but
  CMake did not find one.

  Could not find a package configuration file provided by "NetCDF" with any
  of the following names:

    NetCDFConfig.cmake
    netcdf-config.cmake

  Add the installation prefix of "NetCDF" to CMAKE_PREFIX_PATH or set
  "NetCDF_DIR" to a directory containing one of the above files.  If "NetCDF"
  provides a separate development package or SDK, be sure it has been
  installed.
```

**Comment by SamuelTrahanNOAA** [comment]
I cannot see anything in your branch that would break Jet support, but somehow it fails to compile for me, while the community/develop succeeds.

**Comment by SamuelTrahanNOAA** [comment]
No, cancel that, the develop just failed too. A slightly older version of develop succeeds. I haven't tracked down what broke jet support yet...

I think I need someone else to try, to see if this is something specific to my account.

**Comment by SamuelTrahanNOAA** [comment]
@MinsukJi-NOAA Never mind. Now everything is compiling properly. That may have been a temporary system issue.

**Comment by MinsukJi-NOAA** [comment]
> @MinsukJi-NOAA Never mind. Now everything is compiling properly. That may have been a temporary system issue.

I tried to run the develop on jet, and got this error:
```
+ cp fv3_conf/compile_slurm.IN_jet fv3_conf/compile_slurm.IN
+ mkdir -p /lfs4/HFIP/nems/Minsuk.Ji/RT_BASELINE/Minsuk.Ji
mkdir: cannot create directory '/lfs4/HFIP/nems': Permission denied
++ echo 'rt.sh error on line 396'
rt.sh error on line 396
```

**Comment by SamuelTrahanNOAA** [comment]
Are you setting ACCNR to anything?

**Comment by SamuelTrahanNOAA** [comment]
@MinsukJi-NOAA The develop branch does not run on Jet. You need to use my updates (that's why I did the PR).

**Comment by MinsukJi-NOAA** [comment]
> Are you setting ACCNR to anything?

No. I guess this error will be fixed by your code changes in this PR?

**Comment by SamuelTrahanNOAA** [comment]
Yes. My changes correct the code so it will use h-nems by default on Jet, or the value of ACCNR if you set one. On develop, you had to set ACCNR or it would break.

**Comment by MinsukJi-NOAA** [comment]
It's good to merge this PR into #1009 ?

**Comment by SamuelTrahanNOAA** [comment]
@MinsukJi-NOAA I am testing #1009 and #1016 together right now on Jet. I will let you know when I trust the merge.

/lfs4/HFIP/hfv3gfs/s/pr-1009/tests
/lfs4/HFIP/hfv3gfs/s/rt/r/rt_158888

Edit: I had typed the wrong rt_ directory path.

**Comment by climbfuji** [comment]
Just hardcode ACCNR in rt.sh for testing if develop in principle compiles on jet ...

**Comment by SamuelTrahanNOAA** [comment]
@climbfuji That should still work of course, but with this PR everything will work as originally intended.

**Comment by climbfuji** [comment]
At least I can confirm that there is no problem with finding netcdf, i.e. I don’t have the issue that you reported initially.

**Comment by MinsukJi-NOAA** [comment]
> At least I can confirm that there is no problem with finding netcdf, i.e. I don’t have the issue that you reported initially.

I confirm that there is no issue finding netcdf on jet.

**Comment by SamuelTrahanNOAA** [comment]
I have no trouble finding NetCDF now either. All of the build jobs are succeeding.

**Comment by SamuelTrahanNOAA** [comment]
@MinsukJi-NOAA The Jet tests are going well, so I started Hera intel and gnu. Right now, all three workflows are generating baselines (rt.sh -c)

**Comment by SamuelTrahanNOAA** [comment]
Hera finished generating baselines. It is validating against them now.

/scratch1/NCEPDEV/stmp4/Samuel.Trahan/FV3_RT/REGRESSION_TEST_INTEL
/scratch1/NCEPDEV/stmp4/Samuel.Trahan/FV3_RT/REGRESSION_TEST_GNU

Jet is still generating baselines.

**Comment by SamuelTrahanNOAA** [comment]
Hera gnu and intel tests have passed. Jet has generated its baseline, and is verifying against that now.

/lfs4/HFIP/hfv3gfs/s/rt/t/Samuel.Trahan/FV3_RT/REGRESSION_TEST_INTEL

**Comment by SamuelTrahanNOAA** [comment]
@MinsukJi-NOAA I can confidently say the #1016 and #1009  work together now. However, this PR has not been reviewed yet, and is not scheduled for merge until January 25.

**Comment by MinsukJi-NOAA** [comment]
> @MinsukJi-NOAA I can confidently say the #1016 and #1009 work together now. However, this PR has not been reviewed yet, and is not scheduled for merge until January 25.

Thanks for testing @SamuelTrahanNOAA . Since the changes in this PR are small and help with Regression Testing, this PR can go in with #1009.

**Comment by SamuelTrahanNOAA** [comment]
@MinsukJi-NOAA Someone should review this before it is merged with anything else.

**Review by MinsukJi-NOAA — APPROVED** [review]

**Comment by SamuelTrahanNOAA** [comment]
The Jet tests passed. Though, it doesn't matter much at this point since @MinsukJi-NOAA is merging these changes in his PR #1009. That'll trigger a rerun of the regression test system on all platforms, including Jet.

**Comment by SamuelTrahanNOAA** [comment]
These changes were merged in #1009 so I'm closing this PR.

**Pull request closed without merging**
